### PR TITLE
Iterative & recursive function working

### PR DESCRIPTION
**Pull request by aniolm**
Second part of homework from cpp fundamentals.

### CODE REVIEW

**Comment by ziobron** [comment]
✅ CI Passed - Task: fibonacci
🏅 Granted 6 XP for @aniolm

**Comment by ziobron** [comment]
⏱ 3 XP extra for @aniolm for on-time delivery

File: homework/fibonacci/fibonacci.hpp [hunk 1]
```diff
@@ -1,11 +1,32 @@
 #pragma once
 
 int fibonacci_iterative(int sequence) {
-    // TODO: Your implementation goes here
-    return 0;
+    int Fn_2{0};
+    int Fn_1{1};
```
Comment (nauka-programowania-MA) [inline homework/fibonacci/fibonacci.hpp:5]: What are 'Fn_2' and `Fn_1` ? Variable should always represent real operation.
Like `result` `divider` `multiplier` `error_code` etc...

File: homework/fibonacci/fibonacci.hpp [hunk 2]
```diff
@@ -1,11 +1,32 @@
...
+
+    if(sequence==0){
+        return 0;
+    }else if(sequence==1){
+        return 1;
+    }else{
```
Comment (nauka-programowania-MA) [inline homework/fibonacci/fibonacci.hpp:11]: Add space after while, if, else etc...

File: homework/fibonacci/fibonacci.hpp [hunk 3]
```diff
@@ -1,11 +1,32 @@
...
 
 int fibonacci_iterative(int sequence) {
-    // TODO: Your implementation goes here
-    return 0;
+    int Fn_2{0};
+    int Fn_1{1};
+
+    if(sequence==0){
+        return 0;
+    }else if(sequence==1){
```
Comment (nauka-programowania-MA) [inline homework/fibonacci/fibonacci.hpp:9]: Add space between operations like '+' '-' '!=' '==' etc...

File: homework/fibonacci/fibonacci.hpp [hunk 4]
```diff
@@ -1,11 +1,32 @@
...
+        return 1;
+    }else{
+
```
Comment (nauka-programowania-MA) [inline homework/fibonacci/fibonacci.hpp:12]: Sequence hold exactly the same value as you want to return for 0 and 1 :)

Better write:

`if (sequence < 2) {
    return sequence;
}`


If you return variable here, you don't need to use `else if`

File: homework/fibonacci/fibonacci.hpp [hunk 5]
```diff
@@ -1,11 +1,32 @@
...
+    int Fn_2{0};
+    int Fn_1{1};
+
+    if(sequence==0){
+        return 0;
+    }else if(sequence==1){
+        return 1;
+    }else{
+        
+        for(int i=0;i<sequence;i++)
```
Comment (nauka-programowania-MA) [inline homework/fibonacci/fibonacci.hpp:13]: spaces

File: homework/fibonacci/fibonacci.hpp [hunk 6]
```diff
@@ -1,11 +1,32 @@
...
+    }
 }
 
 int fibonacci_recursive(int sequence) {
-    // TODO: Your implementation goes here
-    return 0;
+    if(sequence==0){
+        return 0;
+    }else if(sequence==1){
+        return 1;
```
Comment (nauka-programowania-MA) [inline homework/fibonacci/fibonacci.hpp:26]: Sequence hold exactly the same value as you want to return for 0 and 1 :)

Better write:

`if (sequence < 2) {
    return sequence;
}`

File: homework/fibonacci/fibonacci.hpp [hunk 7]
```diff
@@ -1,11 +1,32 @@
...
+    }else{
+        return fibonacci_recursive(sequence-2)+fibonacci_recursive(sequence-1);
+    }
```
Comment (nauka-programowania-MA) [inline homework/fibonacci/fibonacci.hpp:29]: If you return variable here, you don't need to use `else if` or `else`

**Review by nauka-programowania-MA — APPROVED** [review]

**Pull request closed without merging**